utils/build/docker/php/parametric/server.php

```diff
@@ -177,6 +157,26 @@ function remappedSpanKind($spanKind) {
     $headers = \DDTrace\generate_distributed_tracing_headers();
     return jsonResponse(["http_headers" => array_map(null, array_keys($headers), $headers)]);
 }));
+$router->addRoute('POST', '/trace/span/extract_headers', new ClosureRequestHandler(function (Request $req) use (&$spans, &$spansDistributedTracingHeaders) {
+    $headers = arg($req, 'http_headers');
+    # Hack: Create a span from the extracted headers to retrieve the parent_id in the distrivuted tracing headers
+    $span = \DDTrace\start_span();
+    $headers = array_merge(...array_map(fn($h) => [strtolower($h[0]) => $h[1]], $headers));
+    $callback = function ($headername) use ($headers) {
+        return $headers[$headername] ?? null;
+    };
+    \DDTrace\consume_distributed_tracing_headers($callback);
+    try {
+        $spanID = $span->parentId;
+    }
+    catch(Error $e) {
+        $spanID = null;
+    }
+    $spansDistributedTracingHeaders[$spanID] = $headers;
+    # do not finish the span, as it is only used to retrieve the parent_id. Delete it to avoid sending it to the agent.
+    unset($span);
```

Comment:
This has no effect. In fact it's not really possible to drop spans, except by having a span on a different span stack and you never close it.
I.e. do a create_stack(), then `switch_stack()` (without arguments) to restore the original stack.

Reply:
I will remove this unset operation. Also do I need to create and switch stacks here? I assumed each request had it's own trace context. 

Also would switching stacks prevent this span from being sent to the agent if `\DDTrace\dd_trace_close_all_spans_and_flush()` is called (this isn't an issue right now, just curious)

Reply:
This server is not instrumented, i.e. no trace stacks are created for requests.

And yes, calling dd_trace_close_all_spans_and_flush() will not include that span then.